browser/brave_ads/ads_service_factory.cc

```diff
@@ -91,14 +91,19 @@ KeyedService* AdsServiceFactory::BuildServiceInstanceFor(
           profile);
 
   std::unique_ptr<AdsServiceImpl> ads_service =
-      std::make_unique<AdsServiceImpl>(
-          profile,
+      std::make_unique<AdsServiceImpl>(profile,
 #if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
-          brave_adaptive_captcha_service,
-          std::make_unique<AdsTooltipsDelegateImpl>(profile),
+                                       brave_adaptive_captcha_service,
+#if BUILDFLAG(IS_ANDROID)
+                                       nullptr,
```

Comment:
Out of interest why do we pass `nullptr` for Android?

Reply:
@tmancey we don't need to use tooltip delegate for android so we are passing nullptr

Reply:
Could we decouple to a function such as `AdsTooltipDelegate`